src/coreclr/nativeaot/Runtime/unix/unixasmmacrosloongarch64.inc

```diff
@@ -0,0 +1,328 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "AsmOffsets.inc"
+
+.macro NESTED_ENTRY Name, Section, Handler
+        LEAF_ENTRY \Name, \Section
+        .ifnc \Handler, NoHandler
+        .cfi_personality 0x1b, C_FUNC(\Handler) // 0x1b == DW_EH_PE_pcrel | DW_EH_PE_sdata4
+        .endif
+.endm
+
+.macro NESTED_END Name, Section
+        LEAF_END \Name, \Section
+.endm
+
+.macro PATCH_LABEL Name
+        .global C_FUNC(\Name)
+C_FUNC(\Name):
+.endm
+
+.macro ALTERNATE_ENTRY Name
+        .global C_FUNC(\Name)
+        .hidden C_FUNC(\Name)
+C_FUNC(\Name):
+.endm
+
+.macro LEAF_ENTRY Name, Section
+        .global C_FUNC(\Name)
+        .hidden C_FUNC(\Name)
+        .type \Name, %function
+C_FUNC(\Name):
+        .cfi_startproc
+.endm
+
+.macro LEAF_END Name, Section
+        .size \Name, .-\Name
+        .cfi_endproc
+.endm
+
+.macro PREPARE_EXTERNAL_VAR Name, HelperReg
+        la.local  \HelperReg, \Name
+.endm
+
+.macro PREPARE_EXTERNAL_VAR_INDIRECT Name, HelperReg
+        la.local  \HelperReg, \Name
+        ld.d  \HelperReg, \HelperReg, 0
+.endm
+
+.macro PREPARE_EXTERNAL_VAR_INDIRECT_W Name, HelperReg
+        la.local  \HelperReg, \Name
+        ld.w  \HelperReg, \HelperReg, 0
+.endm
+
+
+.macro PROLOG_STACK_ALLOC Size
+        addi.d  $sp, $sp, -\Size
+.endm
+
+.macro EPILOG_STACK_FREE Size
+        addi.d  $sp, $sp, \Size
+        .cfi_adjust_cfa_offset -\Size
+.endm
+
+.macro EPILOG_STACK_RESTORE
+        ori  $sp, $fp, 0
+        .cfi_restore  3
+.endm
+
+.macro PROLOG_SAVE_REG reg, ofs
+        st.d  $r\reg, $sp, \ofs
+        .cfi_rel_offset \reg, \ofs
+.endm
+
+.macro PROLOG_SAVE_REG_PAIR reg1, reg2, ofs
+        st.d  $r\reg1, $sp, \ofs
+        st.d  $r\reg2, $sp, \ofs + 8
+        .cfi_rel_offset  \reg1, \ofs
+        .cfi_rel_offset  \reg2, \ofs + 8
+        .ifc  \reg1, $fp
+        ori  $fp, $sp, 0
+        .cfi_def_cfa_register  22
+        .endif
+.endm
+
+.macro PROLOG_SAVE_REG_PAIR_INDEXED reg1, reg2, ofs
+        st.d  $r\reg1, $sp, \ofs
+        st.d  $r\reg2, $sp, \ofs + 8
+        addi.d  $sp, $sp, \ofs
+        .cfi_adjust_cfa_offset  -\ofs
+        .cfi_rel_offset  \reg1, 0
+        .cfi_rel_offset  \reg2, 8
+        .ifc  \reg1, $fp
+        ori  $fp, $sp, 0
+        .cfi_def_cfa_register  $fp
+        .endif
+.endm
```

Comment:
please rewrite this macro by reference [runtime/src/coreclr/pal/inc/unixasmmacrosloongarch64.inc](https://github.com/dotnet/runtime/blame/c07a81f90938a8c3ec5b056b78ddd1718dc9931c/src/coreclr/pal/inc/unixasmmacrosloongarch64.inc#L87)